### PR TITLE
Gps research

### PR DESCRIPTION
**Pull request by hdngr**
Begin researching M9 integration re:  #3811.

A couple of things:
1.  Please provide feedback on if this overall structure looks alright.  I don't think I'm breaking anything, but without any unit testing it is kind of a free for all
2. @GPSFan it would be helpful if you could give some feedback on what the critical configs are to make the M9 functional...  We'll translate them into `32.01`

Feel free to write them as comments in the code here or in #3811.

cc @lolsborn

### CODE REVIEW

**Comment by GPSFan** [comment]
I have an M9 receiver and will be able to test it on my ESP32 based system, but until I get my I/O breakout for my RAK system (on order, shipped) I won't be able to do any testing on an NRF52.
I will review the attached code.

**Comment by hdngr** [comment]
> The hwVersion for the M9 is 90000 not 190000.

That would be logical, wouldn't it.  But its not, lol.

image

**Comment by hdngr** [comment]
> I have an M9 receiver and will be able to test it on my ESP32 based system, but until I get my I/O breakout for my RAK system (on order, shipped) I won't be able to do any testing on an NRF52. I will review the attached code.

Totally.  Just feeding ideas would be helpful.  Have not dug into if they are defaults yet, but seems like setting `RMC` and `GCA` and then turning off all of the other chatty stuff might be the key thing to fix.

**Comment by GPSFan** [comment]
Well, I'll be... shows you what a few years of not using the old M9 does..

File: src/gps/GPS.cpp [hunk 1]
```diff
@@ -646,7 +652,33 @@ bool GPS::setup()
...
+                msglen = makeUBXPacket(0xb5, 0x62, sizeof(_message_1HZ), _message_1HZ);
+                _serial_gps->write(UBXscratch, msglen);
+
+                if (getACK(0x06, 0x08, 500) != GNSS_RESPONSE_OK) {
+                    LOG_WARN("Unable to set GPS update rate.\n");
+                } else {
+                    LOG_INFO("Successfully set GPS update rate.!\n");
+                }
+                delay(1000);
+            }
```
Comment (hdngr) [inline src/gps/GPS.cpp:678]: What other messages should go here?  Not super familiar with all of these protocols.  A dump of the hex code for M9 would be helpful BUT, a pointer directly to config.  Would be helpful.

For instance, what does `_message_VALSET_DISABLE_NMEA_RAM` correspond to?  Would it be [3.10.13 UBX-CFG-NMEA (0x06 0x17)](https://content.u-blox.com/sites/default/files/u-blox-M9-SPG-4.04_InterfaceDescription_UBX-21022436.pdf)

Comment (hdngr) [reply src/gps/GPS.cpp:678]: @GPSFan ^^...  cc: @lolsborn 

@GPSFan if you made comments in here with what configs our commands you think we absolutely need, we can grind them out.  We'd appreciate it! 🙏

**Comment by fifieldt** [comment]
Hi @hdngr @lolsborn , are you still interested in M9 support? I've just merged some changes to the gps code which will make it easier to add M9 specific things. Though, the new code conflicts pretty badly with this pull request, so let's move back to the issue #3811 for discussion?

**Pull request closed without merging**